### PR TITLE
Us18 indicate damage

### PR DESCRIPTION
**Pull request by Wessebanan**
Units now flash white for 50 ms when hit and turn dark based on how much health remains.

UnitColorSwitchSystem resets color and darkens based on health percentage compared to base health.

ColorSwitchEvent takes an entityID, color (uint3) and time (seconds) to switch color for, triggers UnitColorSwitchSystem::onEvent

Moved some stuff from GameLoopSystems.cpp to AIGlobals.h to use some stuff from other places and sync between them (uint3 struct, color defines)

### CODE REVIEW

File: AI/AI/includes/AIGlobals.h [hunk 1]
```diff
@@ -42,4 +42,25 @@ enum MAPINITSETTING
...
-};
\ No newline at end of file
+};
+
+struct uint3
+{
+	unsigned char r, g, b;
+	uint3() : r(0), g(0), b(0) {}
+	uint3(unsigned char r, unsigned char g, unsigned char b) : r(r), g(g), b(b) {}
+};
```
Comment (AmberOrange) [inline AI/AI/includes/AIGlobals.h]: If named after color rgb, why UINT3?!?!"?=!"?=#`!`?¤=0+t5

File: Game/GameMain/Physics/CollisionSystem.cpp [hunk 2]
```diff
@@ -68,17 +68,17 @@ void ecs::systems::ObjectCollisionSystem::onEvent(TypeID _typeID, ecs::BaseEvent
...
+			//{
+			//	DynamicMovementComponent* colliding_movement = getComponentFromKnownEntity<DynamicMovementComponent>(current_entity_id);
+
+			//	// Creating a shove event in the direction of the colliding entities movement.
+			//	ForceImpulseEvent shove;
+			//	shove.mDirection	= p_movement->mDirection;
+			//	shove.mForce		= BASE_KNOCKBACK;
+			//	shove.mEntityID		= current_entity_id;
+			//	createEvent(shove);
+			//}
```
Comment (AmberOrange) [inline Game/GameMain/Physics/CollisionSystem.cpp]: Will this be used anymore?

File: Game/GameMain/Physics/FightingSystem.cpp [hunk 3]
```diff
@@ -322,8 +325,86 @@ void ecs::systems::DamageSystem::updateEntity(FilteredEntity& _entityInfo, float
...
+	uint3 color = p_color_switch->mColor;
+	p_unit_color->red	= color.r;
+	p_unit_color->green = color.g;
+	p_unit_color->blue	= color.b;
+
+	mTimers[p_color_switch->mEntityID] = p_color_switch->mTime;
+}
+void ecs::systems::UnitColorSwitchSystem::updateMultipleEntities(EntityIterator& _entities, float _delta)
+{
+	for (int i = 0; i < _entities.entities.size(); i++)
```
Comment (AmberOrange) [inline Game/GameMain/Physics/FightingSystem.cpp]: Small comment here pls

File: Game/GameMain/Physics/FightingSystem.cpp [hunk 4]
```diff
@@ -294,7 +294,10 @@ void ecs::systems::DamageSystem::updateEntity(FilteredEntity& _entityInfo, float
 		// KNOCKBACK
 		ForceImpulseEvent knockback;
 		knockback.mDirection = getComponentFromKnownEntity<DynamicMovementComponent>(unit_entity->getID())->mDirection;
-		//XMStoreFloat3(&knockback.mDirection, XMVector3Normalize(XMVectorSubtract(XMLoadFloat3(&weapon_bv->GetCenter()), XMLoadFloat3(&weapon_component->mPreviousPos))));
+		knockback.mDirection.y += 0.3f;
```
Comment (AmberOrange) [inline Game/GameMain/Physics/FightingSystem.cpp]: Small comment on this pls? 💯

File: Game/GameMain/Physics/PhysicsEvents.h [hunk 5]
```diff
@@ -35,5 +37,12 @@ namespace ecs
 			float mForce;
 		};
 
+		// Did not know where to put this haha.
```
Comment (AmberOrange) [inline Game/GameMain/Physics/PhysicsEvents.h]: Haha 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣 🤣

File: Game/GameMain/gameGameLoop/GameLoopSystems.cpp [hunk 6]
```diff
@@ -323,38 +323,14 @@ void ecs::systems::RoundStartSystem::readEvent(BaseEvent& event, float delta)
 
 void ecs::systems::RoundStartSystem::CreateUnits()
 {
-	/* TEAM COLORS */
-	struct uint3
+	uint3 army_colors[] =
```
Comment (AmberOrange) [inline Game/GameMain/gameGameLoop/GameLoopSystems.cpp]: Looks cool 🌞

File: Game/GameMain/Physics/CollisionSystem.cpp (hunk 2 above)
Comment (Wessebanan) [reply Game/GameMain/Physics/CollisionSystem.cpp]: skippar shove just nu för det blir knappt någon effekt om jag inte skickar dem uppåt och det blir konstigt

File: Game/GameMain/Physics/FightingSystem.cpp (hunk 3 above)
Comment (Wessebanan) [reply Game/GameMain/Physics/FightingSystem.cpp]: ok

File: Game/GameMain/Physics/FightingSystem.cpp (hunk 4 above)
Comment (Wessebanan) [reply Game/GameMain/Physics/FightingSystem.cpp]: ok

File: Game/GameMain/Physics/PhysicsEvents.h (hunk 5 above)
Comment (Wessebanan) [reply Game/GameMain/Physics/PhysicsEvents.h]: tom hanks

File: Game/GameMain/gameGameLoop/GameLoopSystems.cpp (hunk 6 above)
Comment (Wessebanan) [reply Game/GameMain/gameGameLoop/GameLoopSystems.cpp]: hanku

File: AI/AI/includes/AIGlobals.h (hunk 1 above)
Comment (Wessebanan) [reply AI/AI/includes/AIGlobals.h]: sowwy uwu

**Comment by Wessebanan** [comment]
https://www.google.com/url?sa=i&source=images&cd=&ved=2ahUKEwivsIDv9OTlAhUCIVAKHWZSA8oQjRx6BAgBEAQ&url=https%3A%2F%2Fsupersmashbros.fandom.com%2Fwiki%2FMonado_Arts&psig=AOvVaw1Cfj-i4r_Z9AszfXc9mrCS&ust=1573656347479873

**Pull request merged**